MapboxCoreNavigation/RouteLeg.swift

```diff
@@ -3,6 +3,6 @@ import Turf
 
 extension RouteLeg {
     var coordinates: [CLLocationCoordinate2D] {
-        return (steps.first?.coordinates ?? []) + steps.dropFirst().flatMap { ($0.coordinates ?? []).dropFirst() }
+        return (steps.first?.shape?.coordinates ?? []) + steps.dropFirst().flatMap { ($0.shape?.coordinates ?? []).dropFirst() }
```

Comment:
Consider defining a `+` operator on two LineStrings that concatenates the LineStrings, deduplicating the last coordinate of the first argument and the first argument of the second argument. (This operator could be part of Turf.) Then this method could become something like:

```swift
return steps.dropFirst().compactMap { $0.shape }.reduce(steps.first.shape ?? Polyline())
```

Even better would be to define a `+=` to avoid excessive copying.

Reply:
LineString is a struct, and its `coordinates` property is variable, so it is possible to reduce down into a single LineString.